### PR TITLE
Migrate modern admin theme JS components to TypeScript

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | We're migrating the BO to typescript
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | It's related to migrated forms, so we should verify that every forms are working (even really specific forms)
| Possible impacts? | Every migrated forms functions

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24613)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/components/form/translatable-choice.ts [hunk 1]
```diff
@@ -31,38 +31,50 @@ const {$} = window;
...
-  filterSelect(event) {
+  filterSelect(event: JQueryEventObject): void {
     const $element = $(event.currentTarget);
     const $formGroup = $element.closest('.form-group');
     const language = $element.find('option:selected').val();
 
     // show all the languages selects
-    $formGroup.find(`select.translatable_choice[data-language="${language}"]`).parent().show();
+    $formGroup
+      .find(`select.translatable_choice[data-language="${language}"]`)
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/form/translatable-choice.ts]: Please move to selector map (used at l.60 too)

File: admin-dev/themes/new-theme/js/components/form/translatable-choice.ts [hunk 2]
```diff
@@ -31,38 +31,50 @@ const {$} = window;
 export default class TranslatableChoice {
   constructor() {
     // registers the event which displays the popover
-    $(document).on('change', 'select.translatable_choice_language', (event) => {
-      this.filterSelect(event);
-    });
+    $(document).on(
+      'change',
+      'select.translatable_choice_language',
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/form/translatable-choice.ts]: Please move to selector map

File: admin-dev/themes/new-theme/js/components/form/text-with-recommended-length-counter.ts [hunk 3]
```diff
@@ -52,8 +52,11 @@ export default class TextWithRecommendedLengthCounter {
   constructor() {
     $(document).on('input', '.js-recommended-length-input', (event) => {
       const $input = $(event.currentTarget);
+      const inputVal = <string>$input.val();
 
-      $($input.data('recommended-length-counter')).find('.js-current-length').text($input.val().length);
+      $($input.data('recommended-length-counter'))
+        .find('.js-current-length')
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/form/text-with-recommended-length-counter.ts]: Please move to selector map

File: admin-dev/themes/new-theme/js/components/form/multistore-config-field.ts [hunk 4]
```diff
@@ -30,11 +30,17 @@ export default class MultistoreConfigField {
...
+    $(document).on('change', '.multistore-checkbox', () => {
+      const input = $(this)
+        .closest('.form-group')
+        .find(':input:not(.multistore-checkbox)');
+      const inputContainer = $(this)
+        .closest('.form-group')
+        .find('.input-container');
+      const labelContainer = $(this)
+        .closest('.form-group')
+        .find('.form-control-label');
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/form/multistore-config-field.ts]: Please move to selector map

**Review by matthieu-rolland — COMMENTED** [review]
Except @Progi1984 's feedback, lgtm

**Review by PierreRambaud — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by boubkerbribri — DISMISSED** [review]
UI tests OK.
@Robin-Fischer-PS will check if this needs further checks.

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

You have Conflicting files.

Thanks to check!

**Review by atomiix — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by PululuK — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @NeOMakinG 

Thanks for your perfect work :clap: 
Indeed I took a lot of time to test all the forms, so it's QA :heavy_check_mark: :crossed_fingers: 
All LGTM :rocket: 

Thanks again!

**Comment by Progi1984** [comment]
Thanks @NeOMakinG & @hibatallahAouadni

**Comment by matks** [comment]
@NeOMakinG content for demo 😄

**Comment by matks** [comment]
@hibatallahAouadni thank you for the meticulous QA exploration

**Pull request merged**